### PR TITLE
新增MultiLabelMarginLoss

### PR DESCRIPTION
**Pull request by Qin-sx**
### PR Category

User Experience

### PR Types

Improvements

### Description

**当前问题**  
目前paddle没有MultiLabelMarginLoss

**解决方案**
新增MultiLabelMarginLoss

### CODE REVIEW

**Comment by Qin-sx** [comment]
docs: https://github.com/PaddlePaddle/docs/pull/7331

File: python/paddle/nn/functional/loss.py [hunk 1]
```diff
@@ -4190,6 +4190,171 @@ def multi_margin_loss(
...
+        position_mask = position_indices < first_neg_pos
+
+        # create the final valid mask
+        final_valid_mask = valid_mask & position_mask
+
+        # calculate sample loss
+        sample_loss = paddle.zeros([], dtype=input.dtype)
+
+        # check each position for valid label
+        for j in range(dim):
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/loss.py]: 这里的公式看起来比multi_label_soft_margin_loss要复杂很多，使用的n*n复杂度计算，对性能的影响也比较大，单测超时了。

这里与multi_label_soft_margin_loss的公式差异很大吗，能否有更简洁高效的实现

infoflow 2025-06-23 11-43-38

Comment (Qin-sx) [reply python/paddle/nn/functional/loss.py]: 嗯，收到，因为静态图无法用if else，所以修改了很多，那我改成底层的cpp和cuda实现吧

**Comment by Qin-sx** [comment]
已修改一版新的函数

**Comment by Qin-sx** [comment]
CI报错：
```bash
2025-07-04 08:19:58 /paddle/paddle/phi/kernels/funcs/fft_fill_conj.h(163): error: identifier "ctx" is undefined
2025-07-04 08:19:58           detected during instantiation of "void phi::FFTR2CKernel<T,Context>(const Context &, const phi::DenseTensor &, const std::vector<int64_t, std::allocator<int64_t>> &, const std::__cxx11::string &, __nv_bool, __nv_bool, phi::DenseTensor *) [with T=float, Context=phi::GPUContext]" 
2025-07-04 08:19:58 /paddle/paddle/phi/kernels/gpu/fft_kernel.cu(34): here
2025-07-04 08:19:58 
2025-07-04 08:19:58 /paddle/paddle/phi/kernels/funcs/fft_fill_conj.h(163): error: type name is not allowed
2025-07-04 08:19:58           detected during instantiation of "void phi::FFTR2CKernel<T,Context>(const Context &, const phi::DenseTensor &, const std::vector<int64_t, std::allocator<int64_t>> &, const std::__cxx11::string &, __nv_bool, __nv_bool, phi::DenseTensor *) [with T=float, Context=phi::GPUContext]" 
2025-07-04 08:19:58 /paddle/paddle/phi/kernels/gpu/fft_kernel.cu(34): here
```
应该和pr修改部分无关

猜测可能和这个[pr](https://github.com/PaddlePaddle/Paddle/pull/73813/files#diff-605d8ade5ba1542ac09a5be79e8824ee43c8ae8bc67409256728d42b1c8cbd83)有关

**Comment by Qin-sx** [comment]
PaConvert: https://github.com/PaddlePaddle/PaConvert/pull/599  
Docs: https://github.com/PaddlePaddle/docs/pull/7331

File: test/legacy_test/CMakeLists.txt [hunk 2]
```diff
@@ -1247,3 +1247,4 @@ set_tests_properties(test_imperative_selected_rows_to_lod_tensor
                      PROPERTIES TIMEOUT 200)
 set_tests_properties(test_slice_op PROPERTIES TIMEOUT 120)
 set_tests_properties(test_load_state_dict_from_url PROPERTIES TIMEOUT 40)
+set_tests_properties(test_multimarginloss PROPERTIES TIMEOUT 120)
```
Comment (zhwesky2010) [inline test/legacy_test/CMakeLists.txt]: 目前的API性能如何？这个目前还会超时吗？

Comment (Qin-sx) [reply test/legacy_test/CMakeLists.txt]: 嗯，收到，已修改，等CI测试之后再看一下

**Comment by Qin-sx** [comment]
本地测试时间
```bash
Ran 4 tests in 0.192s
```

**Review by zhwesky2010 — APPROVED** [review]
LGTM
